syntax = "proto3";

package pb;

message SellLog {
    string  id = 1;
    string  crc = 2;
    uint32  emissid = 3;
    string  emissidfull = 4;
    string  cardtype = 5;
    string  lgota = 6;
    uint32  inday = 7;
    uint32  inmonth = 8;
    uint32  inyear = 9;
    uint32  inhour = 10;
    uint32  inmin = 11;
    uint32  insec = 12;
    string  action = 13;
    uint32  actstop = 14;
    uint32  instop = 15;
    uint32  outstop = 16;
    uint32  pricerub = 17;
    uint32  pricekop = 18;
    uint32  psu = 19;
    uint32  tabno = 20;
    uint32  route = 21;
    string  routec = 22;
    uint32  vihod = 23;
    uint32  smena = 24;
    uint32  psutrans = 25;
    uint32  transtype = 26;
    uint32  rideall = 27;
    uint32  tariff = 28;
    uint32  begday = 29;
    uint32  begmonth = 30;
    uint32  begyear = 31;
    uint32  endday = 32;
    uint32  endmonth = 33;
    uint32  endyear = 34;
    uint32  idsell = 35;
    uint32  ctrans = 36;
    uint32  paymentrub = 37;
    uint32  paymentkop = 38;
    string  ext = 39;
    uint32  storno = 40;
    uint32  readflag = 41;
    uint32  outdataday = 42;
    uint32  outdatamonth = 43;
    uint32  outdatayear = 44;
    uint32  outdatahour = 45;
    uint32  outdatamin = 46;
    uint32  outdatasec = 47;
    uint32  crcrec = 48;
    string  saleinfo = 49;
    uint32  sent = 50;
    uint32  confirmed = 51;
    uint32  q_order = 52;
    string  OFD_FN = 53;
    uint32  OFD_FD = 54;
    uint32  OFD_FP = 55;
}

message RidesLog {
    string  id = 1;
    string  crc = 2;
    uint32  route = 3;
    string  routec = 4;
    uint32  vihod = 5;
    uint32  smena = 6;
    uint32  riderest = 7;
    uint32  region = 8;
    uint32  emissid = 9;
    string  emissidfull = 10;
    string  cardtype = 11;
    uint32  town = 12;
    uint32  psu = 13;
    string  lgota = 14;
    uint32  tabno = 15;
    uint32  inday = 16;
    uint32  inmonth = 17;
    uint32  inyear = 18;
    uint32  inhour = 19;
    uint32  inmin = 20;
    uint32  insec = 21;
    uint32  instop = 22;
    uint32  outday = 23;
    uint32  outmonth = 24;
    uint32  outyear = 25;
    uint32  outhour = 26;
    uint32  outmin = 27;
    uint32  outsec = 28;
    uint32  outstop = 29;
    uint32  inflag = 30;
    uint32  pricerub = 31;
    uint32  pricekop = 32;
    uint32  allrub = 33;
    uint32  allkop = 34;
    uint32  cardtrans = 35;
    uint32  zoneflag = 36;
    uint32  psutrans = 37;
    uint32  rideall = 38;
    uint32  psusell = 39;
    uint32  psulast = 40;
    uint32  transtype = 41;
    string  skmocount1 = 42;
    string  skmocount2 = 43;
    string  skmocount3 = 44;
    string  skmocount4 = 45;
    uint32  crcrec = 46;
    uint32  readflag = 47;
    uint32  outdataday = 48;
    uint32  outdatamonth = 49;
    uint32  outdatayear = 50;
    uint32  outdatahour = 51;
    uint32  outdatamin = 52;
    uint32  outdatasec = 53;
    uint32  sent = 54;
    uint32  confirmed = 55;
    uint32  q_order = 56;
    string  ext = 57;
}

message ViewLog {
    string  id = 1;
    string  crc = 2;
    uint32  region = 3;
    uint32  town = 4;
    string  lgota = 5;
    uint32  emissid = 6;
    string  emissidfull = 7;
    string  cardtype = 8;
    string  actcode = 9;
    uint32  actday = 10;
    uint32  actmonth = 11;
    uint32  actyear = 12;
    uint32  acthour = 13;
    uint32  actmin = 14;
    uint32  actsec = 15;
    uint32  actstop = 16;
    uint32  tariffrub1 = 17;
    uint32  tariffkop1 = 18;
    uint32  tariffrub2 = 19;
    uint32  tariffkop2 = 20;
    uint32  psu = 21;
    uint32  tabno = 22;
    uint32  route = 23;
    string  routec = 24;
    uint32  vihod = 25;
    uint32  smena = 26;
    uint32  psulast = 27;
    uint32  psusell = 28;
    uint32  rideall = 29;
    uint32  begday = 30;
    uint32  begmonth = 31;
    uint32  begyear = 32;
    uint32  endday = 33;
    uint32  endmonth = 34;
    uint32  endyear = 35;
    uint32  begday2 = 36;
    uint32  begmonth2 = 37;
    uint32  begyear2 = 38;
    uint32  endday2 = 39;
    uint32  endmonth2 = 40;
    uint32  endyear2 = 41;
    uint32  riderest1 = 42;
    uint32  riderest2 = 43;
    string  skmocount1 = 44;
    string  skmocount2 = 45;
    string  skmocount3 = 46;
    string  skmocount4 = 47;
    uint32  inday = 48;
    uint32  inmonth = 49;
    uint32  inyear = 50;
    uint32  inhour = 51;
    uint32  inmin = 52;
    uint32  insec = 53;
    uint32  outday = 54;
    uint32  outmonth = 55;
    uint32  outyear = 56;
    uint32  outhour = 57;
    uint32  outmin = 58;
    uint32  outsec = 59;
    uint32  psutrans = 60;
    uint32  crcrec = 61;
    uint32  readflag = 62;
    uint32  outdataday = 63;
    uint32  outdatamonth = 64;
    uint32  outdatayear = 65;
    uint32  outdatahour = 66;
    uint32  outdatamin = 67;
    uint32  outdatasec = 68;
    uint32  sent = 69;
    uint32  confirmed = 70;
    uint32  q_order = 71;
    string  ext = 72;
}

message SysEventsLog {
    uint32  psutrans = 1;
    uint32  psu = 2;
    string  actcode = 3;
    uint32  actday = 4;
    uint32  actmonth = 5;
    uint32  actyear = 6;
    uint32  acthour = 7;
    uint32  actmin = 8;
    uint32  actsec = 9;
    string  cardid = 10;
    string  crc = 11;
    uint32  emissid = 12;
    string  emissidfull = 13;
    uint32  tabnumber = 14;
    uint32  gpswide = 15;
    uint32  gpslong = 16;
    uint32  transtype  = 17;
    string  devcode  = 18;
    string  gps_latitude  = 19;
    string  gps_longitude  = 20;
    uint32  gps_valid  = 21;
    uint32  crcrec  = 22;
    uint32  readflag  = 23;
    uint32  outdataday  = 24;
    uint32  outdatamonth  = 25;
    uint32  outdatayear  = 26;
    uint32  outdatahour  = 27;
    uint32  outdatamin  = 28;
    uint32  outdatasec  = 29;
    string  datasys  = 30;
    uint32  sent  = 31;
    uint32  confirmed  = 32;
    uint32  q_order  = 33;
}